### ISSUE TITLE
vulkan - 4.0 dev - cant compile in windows VS 2017, previous version did compile

### ISSUE THREAD

**Issue by rubenjavier** [godotengine/godot#33356]
**Godot version:**
vulkan branch 8b1d5dd from  20191105

**OS/device including version:**
Windows 10 64 bits
VS 2017
python 3.8.0
scons 3.1.1
Intel i7 8750H 16G RAM
Nvidia GTX 1060

**Issue description:**
vulkan branch compiled on windows with previous commit, this one fails almost at the end in math_funcs.h and rasterizer.h

**Steps to reproduce:**
clone vulkan branch from git and compile:

```
git clone https://github.com/godotengine/godot/ --branch vulkan
cd godot
scons platform=windows -j 10 tools=yes target=release_debug
```

Error at the end:

```
[Initial build] Compiling ==> servers\visual\rasterizer_rd\shader_rd.cpp
shader_rd.cpp
servers\visual\rasterizer_rd\rasterizer_scene_forward_rd.cpp(1559): error C2666: 'Math::pow': sobrecargas 2 tienen conversiones similares
[Initial build] C:\Users\Sistemas\Documents\Godot\build\vulkan\godot\core/math/math_funcs.h(101): note: puede ser 'float Math::pow(float,float)'
C:\Users\Sistemas\Documents\Godot\build\vulkan\godot\core/math/math_funcs.h(100): note: o       'double Math::pow(double,double)'Compiling ==> servers\audio\audio_driver_dummy.cpp

servers\visual\rasterizer_rd\rasterizer_scene_forward_rd.cpp(1559): note: al hacer coincidir la lista de argumentos '(float, double)'
audio_driver_dummy.cpp
scons: *** [servers\visual\rasterizer_rd\rasterizer_scene_forward_rd.windows.opt.tools.64.obj] Error 2
C:\Users\Sistemas\Documents\Godot\build\vulkan\godot\servers/visual/rasterizer.h(108): warning C4099: 'RasterizerScene::InstanceBase': nombre de tipo visto anteriormente usando 'struct' ahora se ve usando 'class'
C:\Users\Sistemas\Documents\Godot\build\vulkan\godot\servers/visual/rasterizer.h(98): note: vea la declaración de 'RasterizerScene::InstanceBase'
C:\Users\Sistemas\Documents\Godot\build\vulkan\godot\servers/visual/rasterizer.h(108): warning C4099: 'RasterizerScene::InstanceBase': nombre de tipo visto anteriormente usando 'struct' ahora se ve usando 'class'
C:\Users\Sistemas\Documents\Godot\build\vulkan\godot\servers/visual/rasterizer.h(98): note: vea la declaración de 'RasterizerScene::InstanceBase'
servers\visual\rasterizer_rd\rasterizer_storage_rd.cpp(2619): warning C4804: '>=': uso no seguro del tipo 'bool' en la operación
servers\visual\rasterizer_rd\rasterizer_storage_rd.cpp(2640): warning C4804: '>=': uso no seguro del tipo 'bool' en la operación
servers\visual\rasterizer_rd\rasterizer_storage_rd.cpp(2727): warning C4804: '>=': uso no seguro del tipo 'bool' en la operación
servers\visual\rasterizer_rd\rasterizer_storage_rd.cpp(2750): warning C4804: '>=': uso no seguro del tipo 'bool' en la operación
C:\Users\Sistemas\Documents\Godot\build\vulkan\godot\servers/visual/rasterizer.h(108): warning C4099: 'RasterizerScene::InstanceBase': nombre de tipo visto anteriormente usando 'struct' ahora se ve usando 'class'
C:\Users\Sistemas\Documents\Godot\build\vulkan\godot\servers/visual/rasterizer.h(98): note: vea la declaración de 'RasterizerScene::InstanceBase'
scons: building terminated because of errors.
```

**Comment by rubenjavier** [comment]
recompiling shows the error in a more concise way
```
[Initial build] [94mCompiling [95m==> [93mservers\visual\rasterizer_rd\rasterizer_scene_forward_rd.cpp[0m                                                                                                                                rasterizer_scene_forward_rd.cpp                                                                                                                                                                                                              
[Initial build] C:\Users\Sistemas\Documents\Godot\build\vulkan\godot\servers/visual/rasterizer.h(108): warning C4099: 'RasterizerScene::InstanceBase': nombre de tipo visto anteriormente usando 'struct' ahora se ve usando 'class'         C:\Users\Sistemas\Documents\Godot\build\vulkan\godot\servers/visual/rasterizer.h(98): note: vea la declaración de 'RasterizerScene::InstanceBase'                                                                                            
[Initial build] servers\visual\rasterizer_rd\rasterizer_scene_forward_rd.cpp(1559): error C2666: 'Math::pow': sobrecargas 2 tienen conversiones similares                                                                                    C:\Users\Sistemas\Documents\Godot\build\vulkan\godot\core/math/math_funcs.h(101): note: puede ser 'float Math::pow(float,float)'                                                                                                             C:\Users\Sistemas\Documents\Godot\build\vulkan\godot\core/math/math_funcs.h(100): note: o       'double Math::pow(double,double)'                                                                                                            servers\visual\rasterizer_rd\rasterizer_scene_forward_rd.cpp(1559): note: al hacer coincidir la lista de argumentos '(float, double)'  
```

**Comment by akien-mga** [comment]
Yes Travis and AppVeyor builds seem to be failing.

**Comment by javawag** [comment]
Same issue here, with VS2017 and latest `vulkan` branch. 

The fix is as follows:
```
diff --git a/servers/visual/rasterizer_rd/rasterizer_scene_forward_rd.cpp b/servers/visual/rasterizer_rd/rasterizer_scene_forward_rd.cpp
index 7acbe0598..114e9ecb1 100644
--- a/servers/visual/rasterizer_rd/rasterizer_scene_forward_rd.cpp
+++ b/servers/visual/rasterizer_rd/rasterizer_scene_forward_rd.cpp
@@ -1556,7 +1556,7 @@ void RasterizerSceneForwardRD::_setup_gi_probes(RID *p_gi_probe_probe_cull_resul
                gi_probe_ubo.texture_slot = gi_probe_instance_get_slot(rpi);
                gi_probe_ubo.anisotropy_strength = storage->gi_probe_get_anisotropy_strength(base_probe);
                gi_probe_ubo.ao = storage->gi_probe_get_ao(base_probe);
-               gi_probe_ubo.ao_size = Math::pow(storage->gi_probe_get_ao_size(base_probe),4.0);
+               gi_probe_ubo.ao_size = Math::pow(storage->gi_probe_get_ao_size(base_probe),4.0f);

                if (gi_probe_is_anisotropic()) {
                        gi_probe_ubo.texture_slot *= 3;
```

Basically, add an "f" at the end of line 1559 so that both parameters are `float`s.

**Comment by akien-mga** [comment]
Thanks, I'll push that fix together with code style fixes, looks like @reduz's `clang-format` is not working right now.

**Comment by javawag** [comment]
Awesome! I've always wanted to dip my toe into Godot engine development... didn't imagine it'd be a single character 😜!

**Comment by akien-mga** [comment]
Fixed by #33359.
